src/main/java/duke/sports/MyPlan.java

```diff
@@ -0,0 +1,374 @@
+    }
+
+     /**
+      * View the plan loaded into the current list.
+      * @return the activities in the list.
+      */
+    public String viewPlan() {
+        StringBuilder message = new StringBuilder();
+        int x = 1;
+        for (MyTraining i : getList()) {
+            message.append("Activity ");
+            message.append(x).append(": ").append(i.toString());
+            if (x < getList().size() - 1) {
+                message.append("\n");
+            }
+            x++;
+        }
+        return message.toString();
+    }
+
+     /**
+      * load the plan of specified intensity and value into the list.
+      * @param intensity intensity of plan to be loaded
+      * @param plan plan number passed as a string
+      */
+    public void loadPlan(final String intensity, final String plan) {
+        clearPlan();
+        if (!Intensity.contains(intensity)) {
+            System.out.println("Please input a proper "
+                   + "intensity level: high, moderate, relaxed");
+        } else {
+            int planNum = Integer.parseInt(plan.split("/")[1]);
+            String key = createKey(intensity, planNum);
+            if (getMap().containsKey(key)) {
+                for (MyTraining t : getMap().get(key)) {
+                    getList().add(t);
+                }
+                System.out.println("You have loaded plan " + planNum + " of "
+                        + intensity + " intensity " + " into the list");
+            }
+        }
+    }
+
+     /**
+      * Save the plan in the list to the map if it is edited or newly created.
+      * @param newList List to be saved to map
+      * @param intensity intensity value associated with the plan
+      * @param key key associated with the plan
+      */
+    public void saveToMap(final ArrayList<MyTraining> newList,
+                          final String intensity, final String key) {
+        if (key == "0") {
+            int planNum = 0;
+            Set<String> keys = getMap().keySet();
+            for (String k : keys) {
+                if (k.contains(intensity)) {
+                    planNum++;
+                }
+            }
+            String k = createKey(intensity, planNum);
+            getMap().put(k, newList);
+        } else {
+            getMap().put(key, newList);
+        }
+    }
+
+     /**
+      * Create a plan of specified intensity.
+      * @param intensity intensity of plan to be created.
+      */
+    public void createPlan(final String intensity) {
+        clearPlan();
+        if (Intensity.contains(intensity)) {
+            System.out.println("Creating plan of " + intensity + " intensity.");
+            System.out.println("Please input activity to add in format of "
+                    + "[activity] [number of sets] [number of reps].");
+            while (true) {
+                Scanner sc = new Scanner(System.in);
+                if (sc.hasNextLine()) {
+                    String input = sc.nextLine();
```

Comment:
Should `MyPlan` be scanning user input? This is violating one of the design principles we went through last week.